### DISCUSSION TITLE
how to update file during playback (for streaming)

### DISCUSSION THREAD

**Discussion by Chrysanthemum536** [mpv-player/mpv#14112] [Q&A]
when playing a partially downloaded file, if more of the file is downloaded then the newly downloaded part does not appear until the player is closed and reopened. how to automatically add whatever is downloaded to the video as it is playing, so a video can be streamed without repeatedly reopening the player every few minutes?

**Comment by guidocella** [comment]
It should play new parts even though the UI doesn't update the duration, are you actually trying to play the appended parts? Otherwise there is `appending://PATH`.

**Reply by Chrysanthemum536** [reply]
it still has the problem. but if the video is long enough and I skip to the beginning and back again, refreshing the cache, then the new part is added.

**Reply by Chrysanthemum536** [reply]
I think if a section at the end of the file is downloaded, then it is detected by mpv and everything up to there is already considered cached, and then once the sections in the middle are downloaded they do not get added to the cache so it skips over the middle section on playback

**Reply by Chrysanthemum536** [reply]
if the file is downloaded completely sequentially then it plays properly
